R/deployApp.R

```diff
@@ -345,6 +337,10 @@ deployApp <- function(appDir = getwd(),
   } else {
     displayStatus(paste0(capitalize(assetTypeName), " deployment failed ",
                          "with error: ", response$error, "\n"))
+
+    # a client can still require config with some deployment failures
+    checkForConfig(client, application, showURL, quiet)
```

Comment:
The permission checks occur prior during application creation, prior to bundle upload and deployment.  For example, if you lock a user account and they try to publish, they'll get an error application creation:

```
applicationForTarget(...)
       -> client$createApplication(...)

Error: HTTP 403
POST http://34.209.81.180:3939/__api__/applications
You don't have permission to perform this operation.
```

This code change should only impact cases where the application was created and successfully updated but failed deployment.